SPECS/kubevirt/kubevirt.spec

```diff
@@ -0,0 +1,1670 @@
+Group:          System/Packages
+
+%description    virt-launcher
+The virt-launcher package provides a launcher for kubevirt
+
+%package        virt-operator
+Summary:        Operator component for kubevirt
+Group:          System/Packages
+
+%description    virt-operator
+The virt-opertor package provides an operator for kubevirt CRD
+
+%package        tests
+Summary:        Kubevirt functional tests
+Group:          System/Packages
+
+%description    tests
+The package provides Kubevirt end-to-end tests.
+
+%prep
+%autosetup -p1
+
+%build
+export GOFLAGS="-buildmode=pie"
+KUBEVIRT_VERSION=%{version} \
+KUBEVIRT_SOURCE_DATE_EPOCH="$(date -r LICENSE +%{s})" \
+KUBEVIRT_GIT_COMMIT='v%{version}' \
+KUBEVIRT_GIT_VERSION='v%{version}' \
+KUBEVIRT_GIT_TREE_STATE="clean" \
+build_tests="true" \
+./hack/build-go.sh install \
+    cmd/virt-api \
+    cmd/virt-chroot \
+    cmd/virt-controller \
+    cmd/virt-freezer \
+    cmd/virt-handler \
+    cmd/virt-launcher \
+    cmd/virt-operator \
+    cmd/virt-probe \
+    cmd/virtctl \
+    %{nil}
+
+%install
+mkdir -p %{buildroot}%{_bindir}
+
+install -p -m 0755 _out/cmd/container-disk-v2alpha/container-disk %{buildroot}%{_bindir}/
+install -p -m 0755 _out/cmd/virtctl/virtctl %{buildroot}%{_bindir}/
+install -p -m 0755 _out/cmd/virt-api/virt-api %{buildroot}%{_bindir}/
+install -p -m 0755 _out/cmd/virt-controller/virt-controller %{buildroot}%{_bindir}/
+install -p -m 0755 _out/cmd/virt-chroot/virt-chroot %{buildroot}%{_bindir}/
+install -p -m 0755 _out/cmd/virt-handler/virt-handler %{buildroot}%{_bindir}/
+install -p -m 0555 _out/cmd/virt-launcher/virt-launcher %{buildroot}%{_bindir}/
+install -p -m 0755 _out/cmd/virt-freezer/virt-freezer %{buildroot}%{_bindir}/
+install -p -m 0755 _out/cmd/virt-probe/virt-probe %{buildroot}%{_bindir}/
+install -p -m 0755 _out/cmd/virt-operator/virt-operator %{buildroot}%{_bindir}/
+install -p -m 0755 _out/tests/tests.test %{buildroot}%{_bindir}/virt-tests
+install -p -m 0755 cmd/virt-launcher/node-labeller/node-labeller.sh %{buildroot}%{_bindir}/
+
+# virt-launcher configurations
+mkdir -p %{buildroot}%{_datadir}/kube-virt/virt-launcher
+install -p -m 0644 cmd/virt-launcher/qemu.conf %{buildroot}%{_datadir}/kube-virt/virt-launcher/
+install -p -m 0644 cmd/virt-launcher/libvirtd.conf %{buildroot}%{_datadir}/kube-virt/virt-launcher/
+install -p -m 0644 cmd/virt-launcher/nsswitch.conf %{buildroot}%{_datadir}/kube-virt/virt-launcher/
+
+
+# virt-launcher SELinux policy needs to land in virt-handler container
+install -p -m 0644 cmd/virt-handler/virt_launcher.cil %{buildroot}/
+
+# Install network stuff
+mkdir -p %{buildroot}%{_datadir}/kube-virt/virt-handler
+install -p -m 0644 cmd/virt-handler/nsswitch.conf %{buildroot}%{_datadir}/kube-virt/virt-handler/
+install -p -m 0644 cmd/virt-handler/ipv4-nat.nft %{buildroot}%{_datadir}/kube-virt/virt-handler/
+install -p -m 0644 cmd/virt-handler/ipv6-nat.nft %{buildroot}%{_datadir}/kube-virt/virt-handler/
+
+%files -n kubevirt
```

Comment:
What's the point of this empty package? And you shouldn't need `-n kubevirt`, since it's identical to the `Name` tag's value.

Reply:
Without the empty package, package grapher fails with the error that there is no node for package "kubevirt" and exit. If there is a cleaner way to workaround this problem, please suggest. 
-n forces the package's full name to be kubevirt vs kubevirt-kubevirt (which will cause the build to fail).

Reply:
In my opinion, we could:
`%files` without the `-n kubevirt` ; which would mean the same.
Also, if we want an empty package, lets add a comment that this is desired.
Something like we've done in [qemu.spec](https://github.com/microsoft/CBL-Mariner/blob/2.0/SPECS/qemu/qemu.spec#L1815)

Reply:
Thanks for the suggestion. I have dropped the -n kubevirt and added a comment saying this is deliberate

Reply:
Ah, the famous build error. It's something we have to fix in our tooling but in general specs without a `%files` section for the default package are valid and will build. I'm guessing you've hit that build error when trying to build with the `PACKAGE_BUILD_LIST=kubevirt` argument. In 2.0 the tooling is looking for package names (like `virtctl`), not spec names (like `kubevirt`), so you could locally work around this with `PACKAGE_BUILD_LIST=virtctl`. You can replace `virtctl` with any other subpackage. I tend to omit the `PACKAGE_BUILD_LIST` argument completely and only keep the specs I want to build inside the `SPECS` directory, so a full build captures everything I'm interested in and doesn't waste time on parsing other specs.

**When we use empty package**
We do that to create "metapackages", which don't install anything themselves but have a bunch of run-time dependencies, so we can install a bunch of other packages in one go. A prime example is the `core-packages` package.

In case of `kubevirt` the default package will neither install anything itself nor will it trigger an installation of other packages, so it's just noise we can remove.

Reply:
Thanks for the build tip (I had no idea) and for explaining why empty packages is an overhead. I have removed the empty package line.